sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -567,6 +568,135 @@ class SparkSqlAstBuilder extends AstBuilder {
+   */
+  override def visitCreateVariable(ctx: CreateVariableContext): LogicalPlan = withOrigin(ctx) {
+
+    def format(name: String): String = {
+      if (conf.caseSensitiveAnalysis) name else name.toLowerCase(Locale.ROOT)
+    }
+
+    val multipartIdentifier = visitMultipartIdentifier(ctx.multipartIdentifier)
+    val defaultExpression = if (ctx.variableDefaultExpression() == null) {
+      "null"
+    } else {
+      visitVariableDefaultExpression(ctx.variableDefaultExpression())
+    }
+    val dataTypeStr: Option[String] =
+      if (Option(ctx.dataType).nonEmpty) {
+        Option(source(Option(ctx.dataType).get))
+      } else {
+        Option(null)
+      }
+
+    if (multipartIdentifier.length > 3) {
+      throw QueryParsingErrors.unsupportedVariableNameError(
+        multipartIdentifier, ctx.multipartIdentifier)
+    }
+
+    val schemaQualifiedName = if (multipartIdentifier.length < 2) {
+      SESSION_DATABASE +: multipartIdentifier
+    } else {
+      multipartIdentifier
+    }
+
+    val catalogQualifiedName = if (schemaQualifiedName.length < 3) {
+      SYSTEM_CATALOG +: schemaQualifiedName
+    } else {
+      schemaQualifiedName
+    }
+    val variableIdentifier = VariableIdentifier(catalogQualifiedName)
+
+    if (variableIdentifier != VariableIdentifier(
+      Seq(SYSTEM_CATALOG, SESSION_DATABASE, format(catalogQualifiedName.last)))) {
+      throw QueryParsingErrors.unsupportedVariableNameError(
+        multipartIdentifier, ctx.multipartIdentifier)
+    }
+
+    CreateVariableCommand(variableIdentifier, dataTypeStr, defaultExpression, ctx.REPLACE != null)
+  }
+
+  /**
+   * Create a DROP VARIABLE statement.
+   *
+   * For example:
+   * {{{
+   *   DROP TEMPORARY VARIABLE [IF EXISTS] variable;
+   * }}}
+   */
+  override def visitDropVariable(ctx: DropVariableContext): LogicalPlan = withOrigin(ctx) {
+    val variableName = visitMultipartIdentifier(ctx.multipartIdentifier)
+    if (variableName.length > 3) {
+      throw QueryParsingErrors.unsupportedVariableNameError(variableName, ctx)
+    }
+
+    val variableIdentifier = VariableIdentifier(variableName)
+
+    if (ctx.TEMPORARY() != null) {
+      if (variableIdentifier.database.getOrElse(SESSION_DATABASE) != SESSION_DATABASE ||
+        variableIdentifier.catalog.getOrElse(SYSTEM_CATALOG) != SYSTEM_CATALOG) {
+        throw QueryParsingErrors.unsupportedVariableNameError(variableIdentifier.nameParts, ctx)
+      }
+
+      DropVariableCommand(
+        identifier = VariableIdentifier(Seq(SYSTEM_CATALOG, SESSION_DATABASE,
+          variableIdentifier.variableName)),
+        ifExists = ctx.EXISTS != null)
+    } else {
+      DropVariableCommand(
+        identifier = variableIdentifier,
+        ifExists = ctx.EXISTS != null)
+    }
+  }
+
+  override def visitSetVariable(ctx: SetVariableContext): LogicalPlan = withOrigin(ctx) {
+
+    if (ctx.query() != null) {
+      /**
```

Comment:
We may use the correct one-liner comment based on this information here: https://github.com/databricks/scala-style-guide#documentation-style